temporal-sdk/src/main/java/io/temporal/client/WorkflowStubImpl.java

```diff
@@ -307,65 +304,101 @@ public <R> R update(
       Type resultType,
       Object... args) {
     checkStarted();
-    UpdateOutput<R> result;
+    try {
+      StartUpdateOptions<R> options =
+          StartUpdateOptions.<R>newBuilder()
+              .setUpdateName(updateName)
+              .setUpdateId(updateId)
+              .setFirstExecutionRunId(firstExecutionRunId)
+              .setWaitPolicy(UpdateWaitPolicy.COMPLETED)
+              .setResultClass(resultClass)
+              .setResultType(resultType)
+              .build();
+      return startUpdate(options, args).getResultAsync().get();
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public <R> UpdateHandle<R> startUpdate(String updateName, Class<R> resultClass, Object... args) {
+    StartUpdateOptions<R> options =
+        StartUpdateOptions.<R>newBuilder()
+            .setUpdateName(updateName)
+            .setWaitPolicy(UpdateWaitPolicy.ACCEPTED)
+            .setResultClass(resultClass)
+            .setResultType(resultClass)
+            .build();
+
+    return startUpdate(options, args);
+  }
+
+  @Override
+  public <R> UpdateHandle<R> startUpdate(StartUpdateOptions<R> options, Object... args) {
```

Comment:
I don't want to enumerate every possible type of override for update. `update` should not take `update` because it defeats the purpose of the API

Reply:
Two things:

First, we have `update(String updateName, String updateId, String firstExecutionRunId, Class<R> resultClass, Type resultType, Object... args)`. So can we at least make update and start update overloads consistent?

Second, do we want to encourage the use of update ID? If not, no problem, I agree the update-id overload can be removed. So you have two overloads for each.

Reply:
>First, we have update(String updateName, String updateId, String firstExecutionRunId, Class<R> resultClass, Type resultType, Object... args). So can we at least make update and start update overloads consistent?

At this point in the update feature I don't want `update` to take `UpdateOptions` because `UpdateOptions` only are meant to apply for `StartUpdate`. I would encourage the use of `startUpdate` vs `update`

Reply:
IN the future if we receive user request we can add more overloads for usability

Reply:
Why are update options only meant for start update? Is it just because of wait policy? We can enforce that be unset for the execute one at runtime. We accept workflow options for start or execute workflow, and activity options for start and execute activity. Granted those are called `execute` and `executeAsync`. Maybe we want to change the naming here too to `update` and `updateAsync`?

We should not wait for the future for consistency I don't think, and we shouldn't let wait policy affect the consistency that much. We should simply let you execute or start update with options or with simpler shortcuts, nothing more complicated hopefully.

Reply:
After off-PR discussion here, we can just remove the many-param overload for `update` and make that one sugar `update` be the only sync way and anyone that wants to do anything more advanced needs to use `startUpdate`.